src/code/HTTPRequestHandler/HTTPRequestHandler.cpp

```diff
@@ -44,8 +43,37 @@ void HTTPRequestHandler::GET(HTTPCall &request)
 
 void HTTPRequestHandler::POST(HTTPCall &request)
 {
-	(void)request;
-	request.setResponse("HTTP/1.1 200 OK\r\n\r\nYOU SENT A POST REQUEST");
+	try
+	{
+		const IPath *root(request.getLocation()->getRoot());
+		Path url(root->get() + request.getBasicRequest().getPath());
+		if (!FileManager::isFileExists(url.get()))
+		{
+			ResponseHeader response(HTTPStatusCode(HTTPStatusCode::NOT_FOUND), request.getErrorPages());
+			return (request.setResponse(response.getResponse()));
+		}
+		if (httprequesthandlerPOST::isDirectoryListing(url, request))
+		{
+			ResponseHeader response(HTTPStatusCode(HTTPStatusCode::OK), request.getErrorPages());
+			response.setContentType(httpConstants::HTML_SUFFIX);
+			response.setBody(
+				httprequesthandlerPOST::getDirectoryContent(root, Path(request.getBasicRequest().getPath())));
+			return (request.setResponse(response.getResponse()));
+		}
+		ResponseHeader response(HTTPStatusCode(HTTPStatusCode::OK), request.getErrorPages());
+		response.setBody(httprequesthandlerPOST::getFileContent(url.get(), response));
+		return (request.setResponse(response.getResponse()));
+	}
+	catch (const BasicHTTPRequest::Invalid &e)
```

Comment:
It is not neccesary, the request is supposed to be a right one at this part of the code